airflow/executors/base_executor.py

```diff
@@ -9,16 +9,21 @@
 
 class BaseExecutor(LoggingMixin):
 
-    def __init__(self, parallelism=PARALLELISM):
+    def __init__(self, parallelism=PARALLELISM, env=None):
         """
         Class to derive in order to interface with executor-type systems
         like Celery, Mesos, Yarn and the likes.
 
         :param parallelism: how many jobs should run at one time. Set to
             ``0`` for infinity
         :type parallelism: int
+
+        :param env: environment variables to provide to the child process (
+        defaults to environment of current process)
         """
         self.parallelism = parallelism
+        self.env = env
+
```

Comment:
We add this `env` option to pass custom environment variables to the subprocesses that the executor will launch. The subprocess environment is updated only when `env` is not `None`. We use this functionality to update the `AIRFLOW_CONFIG` environment variable of the subprocesses during tests, without touching the real environment.